checks.d/supervisord.py

```diff
@@ -0,0 +1,158 @@
+from collections import defaultdict
+import errno
+import socket
+import time
+import xmlrpclib
+
+from checks import AgentCheck
+
+import supervisor.xmlrpc
+
+DEFAULT_HOST = 'localhost'
+DEFAULT_PORT = '9001'
+DEFAULT_SOCKET_IP = 'http://127.0.0.1'
+
+DD_STATUS = {
+    'STOPPED': AgentCheck.CRITICAL,
+    'STARTING': AgentCheck.UNKNOWN,
+    'RUNNING': AgentCheck.OK,
+    'BACKOFF': AgentCheck.CRITICAL,
+    'STOPPING': AgentCheck.CRITICAL,
+    'EXITED': AgentCheck.CRITICAL,
+    'FATAL': AgentCheck.CRITICAL,
+    'UNKNOWN': AgentCheck.UNKNOWN
+}
+
+PROCESS_STATUS = {
+    AgentCheck.CRITICAL: 'down',
+    AgentCheck.OK: 'up',
+    AgentCheck.UNKNOWN: 'unknown'
+}
+
+SERVER_TAG = 'supervisord_server'
+
+PROCESS_TAG = 'supervisord_process'
+
+FORMAT_TIME = lambda x: time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(x))
+
+
+class SupervisordCheck(AgentCheck):
+
+    def check(self, instance):
+        server_name = instance.get('name')
+
+        if not server_name or not server_name.strip():
+            raise Exception("Supervisord server name not specified in yaml configuration.")
+
+        supervisor = self._connect(instance)
```

Comment:
Good catch. It actually does, but it does not cause issues because supervisor (the library) is used only in the _connect method. But worth renaming it a different name to avoid future conflicts.